staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -158,7 +157,7 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 				if !ok {
 					_ = rw.send(watch.Event{
 						Type:   watch.Error,
-						Object: &apierrors.NewInternalError(errors.New("retryWatcher: doesn't support resourceVersion")).ErrStatus,
+						Object: &apierrors.NewInternalError(fmt.Errorf("retryWatcher: object %#v doesn't support resourceVersion", event.Object)).ErrStatus,
```

Comment:
maybe %s and dump.Pretty(event.Object) ?

Reply:
Which package provides `dump`? Is that an acceptable dependency for client-go, if it isn't already?

Could it be that the object contains sensitive data, like passwords or access tokens? Such things should not get into errors because errors are likely to be logged.

Reply:
Ignore the question about `dump` - I found that it is "k8s.io/apimachinery/pkg/util/dump".